book/listings/scalable_lists/1/main.rs

```diff
@@ -0,0 +1,96 @@
+mod integer_object;
+
+use gtk::{gio, Label};
+use gtk::{glib::BindingFlags, prelude::*};
+use integer_object::IntegerObject;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.apps_launcher"),
+        Default::default(),
+    );
+
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(app: &gtk::Application) {
+    let window = gtk::ApplicationWindowBuilder::new()
+        .default_width(600)
+        .default_height(600)
+        .application(app)
+        .title("ListView: Applications Launcher")
+        .build();
+
+    let model = gio::ListStore::new(IntegerObject::static_type());
+    for number in 0..100_000 {
+        let integer_object = IntegerObject::from_integer(number);
+        model.append(&integer_object);
+    }
+
+    let factory = gtk::SignalListItemFactory::new();
+    // the "setup" stage is used for creating the widgets
+    factory.connect_setup(move |_factory, item| {
+        let label = Label::new(None);
+        item.set_child(Some(&label));
+    });
+
+    // the bind stage is used for "binding" the data to the created widgets on the "setup" stage
+    factory.connect_bind(move |_factory, list_item| {
+        let integer_object = list_item
+            .item()
+            .unwrap()
+            .downcast::<IntegerObject>()
+            .unwrap();
+
+        let label = list_item.child().unwrap().downcast::<Label>().unwrap();
+        integer_object
+            .bind_property("number", &label, "label")
+            .flags(BindingFlags::SYNC_CREATE)
+            .build();
+    });
```

Comment:
@bilelmoussaoui maybe you can give me a hint, so I don't go in the wrong direction.
Goal: List of integers, if I double click on one of them it increases by 1.
Questions:
- Is binding properties the way to go?
- If yes, what is the typical way to unbind them again (I guess in the unbind step)?